src/ape/api/providers.py

```diff
@@ -291,12 +291,12 @@ def get_transaction(self, txn_hash: str) -> ReceiptAPI:
         """
 
     @abstractmethod
-    def get_transactions_by_block(self, block_id: HexBytes) -> Iterator[TransactionAPI]:
+    def get_transactions_by_block(self, block_id: BlockID) -> Iterator[TransactionAPI]:
         """
         Get the information about a set of transactions from a block.
 
         Args:
-            block_id (HexBytes): The hash of a block.
+            block_id (:class:`~ape.types.BlockID`): The ID of the block.
```

Comment:
This gives us more control, absolutely should have been done this way

Reply:
Ya I saw it was done in the implementation method just not in the abstract. Implementations can always choose not to support a certain ID type too.